examples/fluids/qfunctions/newtonian_state.h

```diff
@@ -404,4 +404,33 @@ CEED_QFUNCTION_HELPER void Vorticity(const State grad_s[3], CeedScalar vorticity
   Curl3(grad_velocity, vorticity);
 }
 
+CEED_QFUNCTION_HELPER int StatePhysicalGradientFromReference(CeedInt Q, CeedInt i, NewtonianIdealGasContext gas, State s, const CeedScalar x_i[3],
+                                                             StateVariable state_var, const CeedScalar *grad_q, const CeedScalar dXdx[3][3],
+                                                             bool zero_dx, State grad_s[3]) {
+  for (CeedInt k = 0; k < 3; k++) {
+    CeedScalar dx_i[3] = {0}, dqi[5];
+    for (CeedInt j = 0; j < 5; j++) {
+      dqi[j] =
+          grad_q[(Q * 5) * 0 + Q * j + i] * dXdx[0][k] + grad_q[(Q * 5) * 1 + Q * j + i] * dXdx[1][k] + grad_q[(Q * 5) * 2 + Q * j + i] * dXdx[2][k];
+    }
+    dx_i[k]   = zero_dx ? 0. : 1.;
+    grad_s[k] = StateFromQ_fwd(gas, s, dqi, x_i, dx_i, state_var);
+  }
+  return CEED_ERROR_SUCCESS;
+}
+
+CEED_QFUNCTION_HELPER int StatePhysicalGradientFromReference_Boundary(CeedInt Q, CeedInt i, NewtonianIdealGasContext gas, State s,
+                                                                      const CeedScalar x_i[3], StateVariable state_var, const CeedScalar *grad_q,
+                                                                      const CeedScalar dXdx[2][3], bool zero_dx, State grad_s[3]) {
+  for (CeedInt k = 0; k < 3; k++) {
+    CeedScalar dx_i[3] = {0}, dqi[5];
+    for (CeedInt j = 0; j < 5; j++) {
+      dqi[j] = grad_q[(Q * 5) * 0 + Q * j + i] * dXdx[0][k] + grad_q[(Q * 5) * 1 + Q * j + i] * dXdx[1][k];
+    }
+    dx_i[k]   = zero_dx ? 0. : 1.;
+    grad_s[k] = StateFromQ_fwd(gas, s, dqi, x_i, dx_i, state_var);
+  }
+  return CEED_ERROR_SUCCESS;
```

Comment:
I think these can return `void` since there is no plausible error path.

Reply:
Yeah, I just did that since a lot of Ratel QFunction helpers I've ~~stolen~~ copied have return values. Not sure if there's any benefit either way.

Reply:
We don't actually use any of the error codes in QFunction helpers currently (if I recall correctly, its because CUDA/HIP backends don't actually check the error codes) but we should probably add some flavor of `CeedCall` for the CPU side of things that expands to nothing for JiT

Reply:
I think of returning an error code like returning `Result` in Rust. If the operation is not supposed to have a "failure" path, then returning `void` is fine. Higher level PETSc operations pretty much all have failure paths so we use the Call macros for uniformity. In purely numerical code, many things should not be able to fail.

Reply:
Ok, I'll remove them for now, just for consistency. And probably remove them from the QFunction helpers.